R/hdi.R

```diff
@@ -246,32 +246,13 @@ hdi.brmsfit <- function(posterior, ci = .90, effects = c("fixed", "random", "all
     ))
   }
 
-  lower <- seq_len(N - window_size)
-  upper <- window_size + lower
+  nCIs <- length(x_sorted) - window_size
+  ci.width <- sapply(1:nCIs, function(.x) x_sorted[.x + window_size] - x_sorted[.x])
 
-  # vectorized difference between edges of cumulative distribution based on scan_length. Values are arranged from left to right scanning.
-  window_width_diff <- x_sorted[upper] - x_sorted[lower]
-
-  # find minimum of width differences, check for multiple minima
-  min_i <- which(window_width_diff == min(window_width_diff))
-  n_candies <- length(min_i)
-
-  if (n_candies > 1) {
-    if (any(diff(sort(min_i)) != 1)) {
-      if (verbose) {
-        warning("HDI: Identical densities found along different segments of the distribution, choosing rightmost.")
-      }
```

Comment:
I _think_ this is an edge case that happens once in ten thousand years...

Reply:
better having it in "just in case"?